src/local.rs

```diff
@@ -163,32 +156,60 @@ pub fn kill_network(node_registry: &NodeRegistry, keep_directories: bool) -> Res
     Ok(())
 }
 
+pub struct LocalNetworkOptions {
+    pub faucet_bin_path: PathBuf,
+    pub join: bool,
+    pub node_count: Option<u16>,
+    pub peers: Option<Vec<Multiaddr>>,
+    pub safenode_bin_path: PathBuf,
+    pub skip_validation: bool,
+}
+
 pub async fn run_network(
     node_registry: &mut NodeRegistry,
-    safenode_bin_path: &Path,
-    faucet_bin_path: &Path,
-    node_count: Option<u16>,
-    skip_validate: bool,
+    service_control: &dyn ServiceControl,
+    network_options: LocalNetworkOptions,
 ) -> Result<()> {
     let launcher = LocalSafeLauncher {
-        safenode_bin_path: safenode_bin_path.to_path_buf(),
-        faucet_bin_path: faucet_bin_path.to_path_buf(),
+        safenode_bin_path: network_options.safenode_bin_path.to_path_buf(),
+        faucet_bin_path: network_options.faucet_bin_path.to_path_buf(),
     };
 
-    let mut port = 12000;
-    let mut rpc_port = 13000;
-    let rpc_client = RpcClient::new(&format!("https://127.0.0.1:{rpc_port}"));
-    let genesis_multiaddr =
-        run_node(port, rpc_port, None, &launcher, node_registry, &rpc_client).await?;
+    let peers = if network_options.join {
+        if let Some(peers) = network_options.peers {
+            peers
+        } else {
+            let peer = node_registry
+                .nodes
+                .iter()
+                .find(|n| n.get_multiaddr().is_some())
+                .ok_or_else(|| eyre!("Unable to obtain a peer to connect to"))?;
+            vec![peer.get_multiaddr().unwrap()]
+        }
+    } else {
+        let port = service_control.get_available_port()?;
+        let rpc_port = service_control.get_available_port()?;
+        let rpc_client = RpcClient::new(&format!("https://127.0.0.1:{rpc_port}"));
+        let genesis_multiaddr = run_node(
+            port,
+            rpc_port,
+            vec![],
+            &launcher,
+            node_registry,
+            &rpc_client,
+        )
+        .await?;
+        vec![genesis_multiaddr]
+    };
 
-    for _ in 2..=node_count.unwrap_or(DEFAULT_NODE_COUNT) {
-        port += 1;
-        rpc_port += 1;
+    for _ in 2..=network_options.node_count.unwrap_or(DEFAULT_NODE_COUNT) {
```

Comment:
By the way, it might be cleaner to provide the default via `clap`:
```
#[clap(long, default_value_t = DEFAULT_NODE_COUNT)]
```

This would mean the default is documented by `clap` (`--help`) and we get rid of an `Option<T>`.